src/Features/Core/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -87,7 +87,10 @@ public State(TService service, SemanticDocument document)
                     return false;
                 }
 
-                this.IsConstant = this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
+                // Interpolated strings are not constants themselves, even if they were composed entirely from constants.
+                this.IsConstant = _service.IsInterpolatedStringExpression(this.Expression)
+                    ? false
+                    : this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
```

Comment:
No, that definitely looks like a bug.  `GetConstantValue()` should only return a value if the compiler is actually going to constant fold the whole expression.